version/version.go

```diff
@@ -61,31 +75,36 @@ func AIsNewerThanB(A, B string) (bool, error) {
 	return false, nil
 }
 
-// RetrieveLatestVersionFromUpstream gets the latest version from the source code at Github
+// RetrieveLatestVersionFromUpstream fetches the latest release version
+// from the GitHub API. The function returns the latest version as a string,
+// or an error if the request failed or the response could not be parsed.
 func RetrieveLatestVersionFromUpstream() (string, error) {
-	resp, err := http.Get("https://raw.githubusercontent.com/getsops/sops/master/version/version.go")
+	req, err := http.NewRequest("GET", "https://api.github.com/repos/getsops/sops/releases/latest", nil)
```

Comment:
The main disadvantage of using this URL is that this can get you quickly rate-limited when calling this often enough from the same IP address.

Alternatively it is possible to `GET https://github.com/getsops/sops/releases/latest/` and extract the version from the `Location` header, **but** that is relying on GitHub not changing the behavior (https://docs.github.com/en/repositories/releasing-projects-on-github/linking-to-releases doesn't document it is a redirect for example, only that you can use the URL to get the latest release), and it did causes problems when the repository is renamed/moved (since it returns a 301 redirect to the other repo's latest URL then). On the other hand, it does not have a rate limit (or at least none I ever reached), similar to https://raw.githubusercontent.com/getsops/sops/...

Or in other words, no matter which method you use, it has drawbacks :)

Reply:
In an initial version I looked at `GITHUB_TOKEN` and `GH_TOKEN` environment variables to try to prevent the rate limiting from happening.

I am aware of the other work around, but worried it may eventually break. Whereas this is a real API, and versioned.

Reply:
Looks good to me! With the fallback it should be pretty robust, assuming that GitHub doesn't do some really crazy things :-)